.github/workflows/my-ci-pipeline.yml

```
@@ -7,13 +7,44 @@ on:
     paths-ignore:
       - '**.md'
 jobs:
- build:
-  name: Build and test
-  runs-on: ubuntu-latest
-  steps:
-    - name: Checkout
-      uses: actions/checkout@v3
-    - name: Build
-      run: docker build --target development --tag todo_app:test .
-    - name: Test
-      run: docker run --env-file ./.env.test --mount type=bind,source="$(pwd)"/todo_app,target=/todo_app/todo_app --mount type=bind,source="$(pwd)"/tests,target=/todo_app/tests --entrypoint poetry todo_app:test run pytest
\ No newline at end of file
+  build:
+    name: Build and test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Build
+        run: docker build --target development --tag todo_app:test .
+      - name: Test
+        run: docker run --env-file ./.env.test --mount type=bind,source="$(pwd)"/todo_app,target=/todo_app/todo_app --mount type=bind,source="$(pwd)"/tests,target=/todo_app/tests --entrypoint poetry todo_app:test run pytest
+  deploy:
+    name: Deploy
+    needs: build
+    runs-on: ubuntu-latest
```

It would be good to add a gate here that let's us dictate when this runs, as we won't usually want it running on every branch/commit like the tests!
```suggestion
    needs: build
    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main'}}
    runs-on: ubuntu-latest
```